Weather App

Alaska01 opened the pull request:
The Weather app is an app built to give temperature for various selected cities. Enter the city and hit the enter on the keyboard. If no city is entered a prompt is displayed asking you to enter a city. You can toggle between ° C or F depending on your preference.

Alaska01 committed:
edit readme/on feature branch

Alaska01 committed:
edit html file

Alaska01 committed:
add weather app html form

Alaska01 committed:
add linters config

Alaska01 committed:
add eslint config

Alaska01 committed:
add stylelintrc config

Alaska01 committed:
fix linter issues

Alaska01 committed:
remove style errors

Alaska01 committed:
add style to project

Alaska01 committed:
add image

Alaska01 committed:
update readme

Alaska01 committed:
edit index file

Alaska01 committed:
edit package.json

Alaska01 committed:
trying to debug webpack

Alaska01 committed:
fix errors with deployment

Alaska01 committed:
add live link

Alaska01 committed:
fix style errors and remove comment

mahmoud717 commented on src/index.js:
```diff
@@ -1 +1,71 @@
-console.log(6);
\ No newline at end of file
+/* eslint-disable no-unused-vars */
```
You should not disable any ESlint rules. instead, try to fix these problems.

mahmoud717 commented on src/index.js:
```diff
@@ -1 +1,71 @@
+let temperatureC = 15;
+
+const searchbox = document.querySelector('.search-box');
+/* eslint-disable no-use-before-define */
```
You should not disable any ESlint rules. instead, try to fix these problems.

mahmoud717 commented on src/index.js:
```diff
@@ -1 +1,71 @@
+const searchbox = document.querySelector('.search-box');
+/* eslint-disable no-use-before-define */
+searchbox.addEventListener('keypress', setQuery);
+function setQuery(evt) {
+  if (evt.keyCode === 13) {
+    if (searchbox.value !== '') {
+      notificationElement.style.display = 'none';
+      getResults(searchbox.value);
+    } else {
+      notificationElement.style.display = 'block';
+      notificationElement.innerHTML = '<p>City Location cannot be empty</p>';
+      notificationElement.style.display = 'flex';
+    }
+  }
+}
+
```
The project requirements specified that you should use ES6 syntax for this project. which means you have to use arrow functions instead of regular functions.

mahmoud717 commented on src/index.js:
```diff
@@ -1 +1,71 @@
+  weatherEl.innerHTML = weather.weather[0].main;
+  const hilow = document.querySelector('.hi-low');
+  hilow.innerText = `${Math.round(weather.main.temp_min)} °C / ${Math.round(weather.main.temp_max)} °C`;
+}
```
The project requirements specified that you should use ES6 syntax for this project. which means you have to use arrow functions instead of regular functions.

mahmoud717 commented on src/index.js:
```diff
@@ -1 +1,71 @@
+  hilow.innerText = `${Math.round(weather.main.temp_min)} °C / ${Math.round(weather.main.temp_max)} °C`;
+}
+
+function dateBuilder(d) {
+  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+
+  const day = days[d.getDay()];
+  const date = d.getDate();
+  const month = months[d.getMonth()];
+  const year = d.getFullYear();
+
+  return `${day} ${date} ${month} ${year}`;
+}
```
The project requirements specified that you should use ES6 syntax for this project. which means you have to use arrow functions instead of regular functions.

mahmoud717 reviewed (commented):
### Changes required ♻️ 

**Hello @Alaska01**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] As requested in the project requirements, the app should change its appearance based on the received data.

- [ ] The toggle button should change the displayed main temp. instead of adding it to the button.

- [ ] check the comments for more details.
- [ ] The commits should be capitalized. make sure to follow that in the upcoming commits.

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

Alaska01 committed:
Fix TSE request

Alaska01 committed:
Add Math.round method

Alaska01 committed:
Delete image folder

Alaska01 committed:
Remove commented code

rukundoeric reviewed (changes requested):
Hi @Alaska01, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  
- [ ]  The app should have  a toggle control to determine if the result will be shown in Celsius or Fahrenheit (There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

Alaska01 committed:
Add project screenshot

Alaska01 committed:
Add mouse hover property in css

Alaska01 commented:
> Hi @Alaska01,
> 
> _Great job so far clap 100 +1. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
> 
> ## Highlights
> * Follows git workflow.  100
> * Codebase looks clean and professional. 100
> * Addressed some of the issues raised from the previous review. 100
> 
> ## Required Changes 
> * [ ]   The app should have  a toggle control to determine if the result will be shown in Celsius or Fahrenheit (There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> * N/A
> 
> Cheers and Happy coding!
> 
> **Feel free to leave any questions or comments in the PR thread if something is not 100% clear.**
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Toggle feature for project already working. Thanks for the review!!

rukundoeric reviewed (approved):
Hi @Alaska01,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] For the better UX(User Experience) It would have been better to add toggle control as a separate button. because there is no way the user can know that you have to click that text in other to switch between C and F.


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

Alaska01 committed:
Add description for temp change

Alaska01 committed:
Add toogle button

Alaska01 committed:
Remove border from button
